Lib/pstats.py

```diff
@@ -470,14 +496,42 @@ def print_call_heading(self, name_size, column_title):
         if subheader:
             print(" "*name_size + "    ncalls  tottime  cumtime", file=self.stream)
 
-    def print_call_line(self, name_size, source, call_dict, arrow="->"):
+    def print_call_line(self, name_size, source, call_dict, arrow="->", sort_tuple=None, sort_type=None, sel_list=()):
         print(func_std_string(source).ljust(name_size) + arrow, end=' ', file=self.stream)
         if not call_dict:
             print(file=self.stream)
             return
-        clist = sorted(call_dict.keys())
+
+        if sort_tuple:
+            stats_list = []
+            calls_only = False
+            for func, value in call_dict.items():
+                if isinstance(value, tuple):
+                    nc, cc, tt, ct = value  # cProfile orders it this way
+                    # matches sort_arg_defs: cc, nc, tt, ct, module, line, name, stdname
+                    stats_list.append((cc, nc, tt, ct) + func +
+                                      (func_std_string(func),))
+                else:
+                    if not calls_only:
+                        if "time" in sort_type:
```

Comment:
```suggestion
if not calls_only and "time" in sort_type:
    ...
```

Reply:
The downside is that this way we will either set `calls_only` to True outside the conditional (set it in every loop) or inside (so we'll check `"time" in sort_type` every loop). I'll go with the former if that's what you prefer.